code/AEPSDK.brs

```diff
@@ -203,24 +203,24 @@ function AdobeAEPSDKInit(taskNode = invalid as dynamic) as object
         '
         ' This function allows passing custom identifiers using identityMap.
         '
-        ' @param data as object : xdm data
+        ' @param data as object : Dictionary containg xdm and freeform data
         ' @param [optional] callback as function(context, result) : handle Edge response
         ' @param [optional] context as dynamic : context to be passed to the callback function
         '
         ' *************************************************************************************
 
-        sendEvent: function(xdmData as object, callback = _adb_defaultCallback as function, context = invalid as dynamic) as void
+        sendEvent: function(data as object, callback = _adb_defaultCallback as function, context = invalid as dynamic) as void
             _adb_logDebug("API: sendEvent()")
-            if _adb_isEmptyOrInvalidMap(xdmData) then
-                _adb_logError("sendEvent() - Cannot send event, invalid XDM data")
+
+            if _adb_isEmptyOrInvalidMap(data.xdm) then
+                _adb_logError("sendEvent() - Cannot send event, XDM data not found.")
                 return
             end if
-            ' event data: { "xdm": xdmData }
+
+            ' event data: { "xdm": xdmData, "data" : data }
             ' add a timestamp to the XDM data
-            xdmData.timestamp = _adb_ISO8601_timestamp()
-            event = _adb_RequestEvent(m._private.cons.PUBLIC_API.SEND_EDGE_EVENT, {
-                xdm: xdmData,
-            })
+            data.xdm.timestamp = _adb_ISO8601_timestamp()
```

Comment:
in the mobile SDK we check if timestamp is provided and we don't override, is this a requirement for roku too?

Reply:
This to have the ts in the required format, since there is no easy way to convert ts in millis to ISO8601, so we do it at the source.